### PR TITLE
TSK-1582: Additional statistical data from monitoring component

### PR DESCRIPTION
**Pull request by gitgoodjhe**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [x] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by gitgoodjhe** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1582&id=gitgoodjhe_taskana

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  @Test
+  void
+      should_ReturnAllOpenTasksByState_When_QueryingForSpecificWorkbasketAndReadyAndClaimedState() {
+
+    HttpEntity<String> request = new HttpEntity<>(restHelper.getHeadersAdmin());
+    ResponseEntity<ReportRepresentationModel> response =
+        TEMPLATE.exchange(
+            restHelper.toUrl(RestEndpoints.URL_MONITOR_TASKS_STATUS_REPORT)
+                + "?workbasket-ids=WBI:100000000000000000000000000000000007&states=READY,CLAIMED",
```
[the remark this answers is not in the document]
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:35]: I have no idea why your version works.. It should'nt. As our REST documentation states we should always expand the query parameters ;)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    int[] tasksPerState = response.getBody().getSumRow().get(0).getCells();
+    // should be 2 READY, 13 CLAIMED
+    int[] expectedTasksPerState = new int[] {2, 13};
+    assertThat(tasksPerState).isEqualTo(expectedTasksPerState);
+  }
+
+  @Test
+  void should_ReturnAllOpenTasksByState_When_QueryingForSpecificWbAndStateReadyAndMinimumPrio() {
+
+    HttpEntity<String> request = new HttpEntity<>(restHelper.getHeadersAdmin());
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:54]: something for the CoP? We don't do this equally. Typically we inline this variable. But I've found some uses which do that like you did.

File: lib/taskana-core/src/test/java/acceptance/report/ProvideTaskStatusReportAccTest.java [hunk 3]
```diff
@@ -190,6 +190,26 @@ void testCompleteTaskStatusReportWithStates() throws Exception {
...
+    assertThat(report).isNotNull();
+    assertThat(report.rowSize()).isEqualTo(2);
+
+    Row<TaskQueryItem> row1 = report.getRow("USER-1-1");
+    assertThat(row1.getCells()).isEqualTo(new int[] {3, 1, 0, 0, 0});
+    assertThat(row1.getTotalValue()).isEqualTo(4);
+
+    Row<TaskQueryItem> sumRow = report.getSumRow();
+    assertThat(sumRow.getCells()).isEqualTo(new int[] {3, 2, 0, 0, 0});
+    assertThat(sumRow.getTotalValue()).isEqualTo(5);
```
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/report/ProvideTaskStatusReportAccTest.java:210]: I especially modified the priority for these tasks so yes, of course this is checked

File: lib/taskana-core/src/main/java/pro/taskana/monitor/internal/reports/TaskStatusReportBuilderImpl.java [hunk 4]
```diff
@@ -42,17 +43,16 @@ public TaskStatusReport buildReport() throws NotAuthorizedException {
...
               () ->
-                  workbasketService
-                      .createWorkbasketQuery()
+                  workbasketService.createWorkbasketQuery()
                       .keyIn(report.getRows().keySet().toArray(new String[0]))
-                      .domainIn(domains != null ? domains.toArray(new String[0]) : null)
-                      .list()
+                      .domainIn(domains != null ? domains.toArray(new String[0]) : null).list()
                       .stream()
                       .collect(
```
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply lib/taskana-core/src/main/java/pro/taskana/monitor/internal/reports/TaskStatusReportBuilderImpl.java:57]: True, I imported the qa/intellij/taskana_formatter_intellij.xml but it still formats like this

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import pro.taskana.common.rest.RestEndpoints;
+import pro.taskana.common.test.rest.RestHelper;
+import pro.taskana.common.test.rest.TaskanaSpringBootTest;
+import pro.taskana.monitor.rest.models.ReportRepresentationModel;
+
+/** Test MonitorController. */
+@TaskanaSpringBootTest
+class MonitorControllerIntTest {
+
+  static RestTemplate template = RestHelper.TEMPLATE;
```
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:24]: Done

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  @Autowired RestHelper restHelper;
+
+  @Test
+  void
+      should_ReturnAllOpenTasksByState_When_QueryingForSpecificWorkbasketAndReadyAndClaimedState() {
```
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:29]: Suggestions?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:35]: Magic

File: rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java (hunk 2 above)
Comment (gitgoodjhe) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/monitor/rest/MonitorControllerIntTest.java:54]: sure

File: lib/taskana-core/src/main/java/pro/taskana/monitor/internal/reports/TaskStatusReportBuilderImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/monitor/internal/reports/TaskStatusReportBuilderImpl.java:57]: weird.. My formatting doesn't change that.. Let's merge that for now.. Will be fixed soon anyway.

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**